src/BusinessLayer/TelemetryReporting.h

```diff
@@ -14,32 +20,52 @@ class TelemetryReporting : public QObject
 {
     Q_OBJECT
 public:
-   TelemetryReporting(CommunicationService& commService,
-                      VehicleData& vehicleData,
-                      View& view);
+    TelemetryReporting(CommunicationService& commService,
+                       KeyMotorData& keyMotorData_,

```

Comment:
All data references passed in should be const. TelemetryReporting should not be able to modify anything as it just reports.